### PR TITLE
feat(proof-of-concept): GraphQL subscription for operations

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Basic proof of concept for using GraphQL subscriptions for operations
 - A bit rough and ready but demonstrates the idea!

## Next steps...
- Remove / hide component to merge
- Look into whole page re-rendering when modal opens - this is not unique to this component and already happens

https://github.com/theopensystemslab/planx-new/assets/20502206/bf9f5711-e83d-4957-a3ca-7dec06e391bb

### CODE REVIEW

File: editor.planx.uk/src/lib/graphql.ts [hunk 1]
```diff
@@ -34,11 +37,62 @@ const customFetch = async (
...
+  };
+});
+
+const authHttpLink = authMiddleware.concat(httpLink);
+
+/**
+ * Authenticated web socket connection - used for GraphQL subscriptions
+ */
+const authWsLink = new WebSocketLink({
+  uri: `ws://localhost:7100/v1/graphql`,
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/lib/graphql.ts:65]: might be convenient/consistent to add this to editor .env (not secret!) as: 
```
REACT_APP_HASURA_WEBSOCKET=ws://localhost:7100/v1/graphql
```

File: editor.planx.uk/src/lib/graphql.ts [hunk 2]
```diff
@@ -34,11 +37,62 @@ const customFetch = async (
...
+  ({ query }) => {
+    const definition = getMainDefinition(query);
+    return (
+      definition.kind === "OperationDefinition" &&
+      definition.operation === "subscription"
+    );
+  },
+  authWsLink,
+  authHttpLink,
+);
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/lib/graphql.ts:94]: :+1:

**Review by jessicamcinchak — APPROVED** [review]
Very cool - don't see any harm in this merging! 

Excited to add some of my operations formatting work on top of this :slightly_smiling_face:

File: editor.planx.uk/src/lib/graphql.ts (hunk 1 above)
Comment (DafyddLlyr) [reply editor.planx.uk/src/lib/graphql.ts:65]: Good spot - had intended to fix this! I'll add a variable for this now.

Comment (DafyddLlyr) [reply editor.planx.uk/src/lib/graphql.ts]: Fixed in [456eaed](https://github.com/theopensystemslab/planx-new/pull/3002/commits/456eaed695d00d6d422115074e4c6ac4dc663097)

Comment (DafyddLlyr) [reply editor.planx.uk/src/lib/graphql.ts]: Just checked this out - we don't actually need to update `.env.example` here as this is an env var within the React app only (and it's a genuine env var, not a secret).

**Pull request merged**